clippy_lints/src/no_effect.rs

```diff
@@ -98,20 +98,6 @@ fn has_no_effect(cx: &LateContext<'_, '_>, expr: &Expr) -> bool {
                     false
                 }
         },
-        ExprKind::Assign(ref left, ref right) => {
```

Comment:
Removing this match arm will change the behavior of this function, since it will return `false` for every assignment, while as of now it will return `true` if both the LHS and RHS don't have an effect.

---
Unrelated: Since this function was implemented, there were more variants added to the `ExprKind` enum. Maybe someone should take a look at this function, remove the wildcard match and take care of the other cases explicitly.

Reply:
Yes, and I want specifically that, moving the list to `temporary_assignment`. I added this branch a few days ago anyways.

Reply:
Ah ok, this makes sence!